src/v/kafka/server/datalake_throttle_manager.cc

```diff
@@ -203,29 +227,23 @@ std::chrono::milliseconds
 datalake_throttle_manager::calculate_throttle() const {
     vassert(
       ss::this_shard_id() == 0, "Throttle can only be calculate on shard 0");
-    if (_translation_status.partitions_translation_blocked > 0) {
-        // if translation is blocked (this should never really be the case),
-        // use max throttle
-        return _max_kafka_throttle();
-    }
+
     // Heuristic:
     // we apply the incremental throttle based on time spent in the degraded
     // translation state, the longer the time the more we throttle. The
     // throttle is set to max after 5 minutes
-
-    if (_translation_status.overdue_translation_partition_count > 0) {
-        auto time_in_degraded_state
-          = std::chrono::duration_cast<std::chrono::milliseconds>(
-            ss::lowres_clock::now() - _last_no_issues_timestamp);
-
-        const auto max_throttle_ratio = std::min(
-          double(time_in_degraded_state.count()) / 300000, 1.0);
-
-        return std::chrono::milliseconds(static_cast<size_t>(
-          max_throttle_ratio * _max_kafka_throttle().count()));
+    if (!needs_throttling()) {
+        return no_throttling;
     }
+    auto time_in_degraded_state
+      = std::chrono::duration_cast<std::chrono::milliseconds>(
+        ss::lowres_clock::now() - _last_no_issues_timestamp);
+
+    const auto max_throttle_ratio
+      = std::min(double(time_in_degraded_state.count()) / 3000000, 1.0) * 0.1;
```

Comment:
What is this 0.1 multiplier for? Could you add a comment mentioning it?

Reply:
Will add a comment, in general i wanted to make the throttle ramp up a bit slower

Reply:
Got it, thanks!